service/src/lib.rs

```diff
@@ -57,37 +42,27 @@ pub trait Factory: Send + Sync {
 /// of custom resources, please [get in touch](https://discord.gg/shuttle) and detail your use case. We'll be interested to see what you
 /// want to provision and how to do it on your behalf on the fly.
 #[async_trait]
-pub trait ResourceBuilder: Default {
-    /// The type of resource this plugin creates.
-    /// If dealing with a Shuttle-provisioned resource, such as a database, use the corresponding variant.
-    /// Otherwise, use the `Custom` variant.
-    const TYPE: resource::Type;
-
-    /// The input config to this resource.
-    type Config: Serialize;
-
-    /// The output from requesting this resource.
-    /// A cached copy of this will be used if the same [`ResourceBuilder::Config`] is found for this [`ResourceBuilder::TYPE`].
-    type Output: Serialize + DeserializeOwned;
-
-    /// Get the config of this plugin after it has been built from its macro arguments with the builder pattern.
+pub trait IntoResourceInput: Default {
```

Comment:
We should keep this as `ResourceBuilder` and double down on naming the method `build()`

Reply:
It does not build a resource, it (builds and) emits a request(+config) for a resource.
`IntoResourceRequest` is not good since it can also do a non-request (input := request | custom struct)

Reply:
`ResourceConfigBuilder` then :smile:

Reply:
How bout `ResourceInputBuilder` and `build() -> Result<Self::Input...`? :pleading_face: